src/modules/sensors/als.c

```diff
@@ -146,8 +146,12 @@ static int capture(void *dev, double *pct, const int num_captures, char *setting
             }
         }
 
-        if (illuminance >= 0) {
-            pct[ctr++] = illuminance / max;
+        if (illuminance >= 1) {
+            if (illuminance != min) {
+                pct[ctr++] = (log10(illuminance) - log10(min)) / log10(max);
```

Comment:
What about just using:
`pct[ctr++] = log10(illuminance) / log10(max)` ?
Ie: with your algorithm you are not using full scale (ie: from ALS_ILL_MIN to ALS_ILL_MAX) but just a subset (with your default values, you can only reach around 0.75: (5 - 1.30) / 5).

Reply:
You're right; I jumped the gun a little on trying to scale it properly. Will make the proper adjustments